core/src/main/java/org/jdbi/v3/wip/categories/OptionalPrimitivesPlugin.java

```diff
@@ -0,0 +1,80 @@
+package org.jdbi.v3.wip.categories;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.Types;
+import java.util.Optional;
+import java.util.OptionalDouble;
+import java.util.OptionalInt;
+import java.util.OptionalLong;
+import org.jdbi.v3.core.Jdbi;
+import org.jdbi.v3.core.generic.GenericType;
+import org.jdbi.v3.core.spi.JdbiPlugin;
+
+import static org.jdbi.v3.wip.categories.PluginUtils.registerClassArgument;
+import static org.jdbi.v3.wip.categories.PluginUtils.registerClassMapper;
+import static org.jdbi.v3.wip.categories.PluginUtils.registerGenericTypeArgument;
+import static org.jdbi.v3.wip.categories.PluginUtils.registerGenericTypeMapper;
+
+public class OptionalPrimitivesPlugin implements JdbiPlugin {
+	@Override
+	public void customizeJdbi(Jdbi jdbi) {
+		registerOptionalArgument(jdbi, new GenericType<Optional<Boolean>>() {}, Types.BOOLEAN, PreparedStatement::setBoolean);
```

Comment:
The existing `Optional` handling for arguments and column mapping takes the `T` parameter (from `Optional<T>` and just looks up an argument / column mapper for type `T`, and delegates to that argument or mapper.